tests/pytests/test_multithread.py

```diff
@@ -363,3 +363,69 @@ def test_multiple_loaders():
     cmd += ['LIMIT', '0', limit]
 
     env.expect(*cmd).noError().apply(lambda x: x[1:]).equal([['n', '0'], ['n', '1'], ['n', '2'], ['n', '3'], ['n', '4']])
+
+@skip(cluster=True)
+def test_switch_loader_modes():
+    # Create an environment with MT_MODE_FULL (0)
+    env = initEnv('WORKER_THREADS 1 MT_MODE MT_MODE_FULL')
+    n_docs = 10
+    cursor_count = 2
+    # Having two loaders to test when the loader is last and when it is not
+    query = ('FT.AGGREGATE', 'idx', '*', 'LOAD', '*', 'LOAD', '*', 'WITHCURSOR', 'COUNT', cursor_count)
+    read_from_cursor = lambda cursor: env.expect('FT.CURSOR', 'READ', 'idx', cursor).noError().res[1]
```

Comment:
consider adding some validation on the results (the results themselves or their count)

Reply:
I decided not to because I wanted to delete the cursor and not run it until depletion. The results themselves are not interesting enough for validation (any corruption should be caught by ASAN or VG)